Add a keyboard shortcut for beautify request body

eimanip opened the pull request:
<!--
Please open an [Issue](https://github.com/kong/insomnia/issues/new) first to discuss new
features or non-trivial changes. Please provide as much detail as possible on the change as
possible including general description, implementation details, potential shortcommings, etc.

If this PR closes an issue, please mention "Closes #XX" where #XX is the issue number.
-->

I set the new shortcut to `Shift+Ctrl+I`. It's the same keyboard shortcut for "Format Document" in VSCode, so I guess it could be a good candidate. Also, most of the current project's hotkeys follow the `Shift+Ctrl+[Char]` pattern. Feel free to suggest another default shortcut for this feature.

This pull request closes #1851 (partially), closes #1915 .

![Screenshot from 2020-10-17 01-53-38](https://user-images.githubusercontent.com/20307301/96317639-caf2c000-101b-11eb-9fb5-d39c1b661a7f.png)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kong/insomnia?pullRequest=2733) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Kong/insomnia?pullRequest=2733) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kong/insomnia?pullRequest=2733) it.</sub>

eimanip committed:
Add a keyboard shortcut (Shift+Ctrl+I) for beautify request body

dimitropoulos committed:
removes requirement for KeydownBinder to wrap

there's no reason for it to, it's using a global listener (well, `document.body`, but still..)

dimitropoulos committed:
updates copy of shortcut

since this applies to more than just request bodies, I thought it best to keep it generic.  It applies to the timeline, response body, test code, etc.

dimitropoulos committed:
removes duplicate logic and fixes render skipping

I noticed a bug that didn't show up before because there was previously no way to beautify (and, thus, trigger this code path) for documents that weren't showing the beautify button.  Now, since the keyboard shortcut will trigger it for everything, what can happen is that the viewport and cursor will reset even if there's nothing to change.  The `_handleBeautify` method before doesn't serve much purpose since all it does is call `_prettify` (and, it was the only thing calling `_prettify`).  Before doing that, though, it's worthless to bother setting the value.  This way, if the document cannot be prettified then it won't be.

dimitropoulos committed:
formats curl codemirror extension

I noticed that the comments were messed up (from the flow-to-ts conversion) here.  change it to self documenting code and touched no logic whatsoever.

gatzjames reviewed (approved):
Works great! Rebased to merge conflicts.
